### ISSUE TITLE
WARNING with 0.6.5.10 and kernel 4.12.0

### ISSUE THREAD

**Issue by yasij** [zfsonlinux/zfs#6324]
### System information

Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | Kubuntu
Distribution Version    | 17.04
Linux Kernel                 | 4.12.0
Architecture                 | x86_64
ZFS Version                  | 0.6.9.10-1ubuntu1
SPL Version                  | 0.6.9.10-1

### Describe the problem you're observing
I'm getting a warning on boot during zfs.mount with 0.6.9.10 and kernel 4.12.0. Kernel log for the WARNING is posted below.
The warning triggered is in super_setup_bdi_name:
WARN_ON(sb->s_bdi != &noop_backing_dev_info);

It looks like the backport commit for 4.12 support from #6089 missed a piece of the commit in master. [e4cb6ee6 commit in 0.6.9.10](https://github.com/zfsonlinux/zfs/commit/e4cb6ee6a5f52b5ee8bca73718a6a57ef23fab88)
[7dae2c81 commit in master](https://github.com/zfsonlinux/zfs/commit/7dae2c81e7b2e68a596c5b431444be0fae308156)

I've attached a patch with the missing change and confirmed the warning no longer appears:
[2200-Linux-4.12-compat-super_setup_bdi_name-fix.txt](https://github.com/zfsonlinux/zfs/files/1131314/2200-Linux-4.12-compat-super_setup_bdi_name-fix.txt)

### Describe how to reproduce the problem
Mount a zfs partition on 4.12.0 and 0.6.9.10.
### Include any warning/errors/backtraces from the system logs
```
Jul  7 00:22:10 amypond kernel: [    9.981526] ------------[ cut here ]------------
Jul  7 00:22:10 amypond kernel: [    9.981537] WARNING: CPU: 4 PID: 2206 at fs/super.c:1281 super_setup_bdi_name+0xbe/0xd0
Jul  7 00:22:10 amypond kernel: [    9.981538] Modules linked in: vfat fat arc4 snd_hda_codec_hdmi snd_hda_codec_realtek snd_hda_codec_generic zfs(PO) zunicode(PO) zavl(PO) zcommon(PO) znvpair(PO) snd_hda_intel snd_hda_codec snd_hwdep snd_hda_core snd_pcm_oss snd_mixer_oss spl(O) snd_pcm snd_seq_oss snd_seq_midi snd_rawmidi coretemp intel_rapl x86_pkg_temp_thermal intel_powerclamp crct10dif_pclmul crc32_pclmul crc32c_intel ghash_clmulni_intel pcbc snd_seq_midi_event snd_seq aesni_intel snd_seq_device aes_x86_64 crypto_simd iwlmvm snd_timer glue_helper efi_pstore mac80211 cryptd intel_cstate intel_rapl_perf efivars snd iwlwifi soundcore cfg80211 cdc_acm shpchp sg btusb btrtl hci_uart btbcm serdev btqca btintel bluetooth battery ecdh_generic rfkill intel_lpss_acpi intel_lpss mfd_core tpm_crb tpm_tis acpi_pad acpi_als tpm_tis_core kfifo_buf
Jul  7 00:22:10 amypond kernel: [    9.981618]  tpm industrialio nfsd auth_rpcgss oid_registry nfs_acl lockd grace sunrpc efivarfs ip_tables x_tables ipv6 crc_ccitt hid_generic usbhid igb e1000e dca ptp psmouse pps_core i915 hwmon intel_gtt i2c_algo_bit drm_kms_helper xhci_pci syscopyarea sysfillrect xhci_hcd sysimgblt fb_sys_fops usbcore drm usb_common fan thermal video i2c_hid pinctrl_sunrisepoint pinctrl_intel i2c_core button
Jul  7 00:22:10 amypond kernel: [    9.981698] CPU: 4 PID: 2206 Comm: mount.zfs Tainted: P           O    4.12.0-customskl #1
Jul  7 00:22:10 amypond kernel: [    9.981701] Hardware name: Gigabyte Technology Co., Ltd. H270N-WIFI/H270N-WIFI-CF, BIOS F5 04/24/2017
Jul  7 00:22:10 amypond kernel: [    9.981704] task: ffff961ed9556a00 task.stack: ffffb7a3ca9a4000
Jul  7 00:22:10 amypond kernel: [    9.981714] RIP: 0010:super_setup_bdi_name+0xbe/0xd0
Jul  7 00:22:10 amypond kernel: [    9.981716] RSP: 0018:ffffb7a3ca9a7d28 EFLAGS: 00010287
Jul  7 00:22:10 amypond kernel: [    9.981719] RAX: 0000000000000000 RBX: ffff961ecc3eb800 RCX: 0000000000000002
Jul  7 00:22:10 amypond kernel: [    9.981721] RDX: 0000000000000001 RSI: 00000000fffffe00 RDI: ffffffffa4166f4c
Jul  7 00:22:10 amypond kernel: [    9.981723] RBP: ffffb7a3ca9a7da0 R08: ffff961ed5e0fab8 R09: 0000000000000000
Jul  7 00:22:10 amypond kernel: [    9.981725] R10: ffffe54310303c00 R11: 0000000000000069 R12: ffff961edada8800
Jul  7 00:22:10 amypond kernel: [    9.981727] R13: ffffffffc0b74a46 R14: ffffb7a3ca9a7db0 R15: 0000000000000000
Jul  7 00:22:10 amypond kernel: [    9.981730] FS:  00007fecf836c780(0000) GS:ffff961eeed00000(0000) knlGS:0000000000000000
Jul  7 00:22:10 amypond kernel: [    9.981732] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Jul  7 00:22:10 amypond kernel: [    9.981734] CR2: 0000564aebdd0868 CR3: 0000000418943000 CR4: 00000000003406e0
Jul  7 00:22:10 amypond kernel: [    9.981737] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
Jul  7 00:22:10 amypond kernel: [    9.981740] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
Jul  7 00:22:10 amypond kernel: [    9.981741] Call Trace:
Jul  7 00:22:10 amypond kernel: [    9.981796]  ? dsl_pool_rele+0x19/0x30 [zfs]
Jul  7 00:22:10 amypond kernel: [    9.981844]  ? rrw_exit+0xc2/0x140 [zfs]
Jul  7 00:22:10 amypond kernel: [    9.981884]  ? zfs_domount+0xd5/0x350 [zfs]
Jul  7 00:22:10 amypond kernel: [    9.981916]  ? zpl_kill_sb+0x20/0x20 [zfs]
Jul  7 00:22:10 amypond kernel: [    9.981953]  ? zpl_fill_super+0x25/0x40 [zfs]
Jul  7 00:22:10 amypond kernel: [    9.981958]  ? mount_nodev+0x46/0xa0
Jul  7 00:22:10 amypond kernel: [    9.981964]  ? mount_fs+0x36/0x140
Jul  7 00:22:10 amypond kernel: [    9.981971]  ? vfs_kern_mount+0x62/0x100
Jul  7 00:22:10 amypond kernel: [    9.981976]  ? do_mount+0x1a7/0xc20
Jul  7 00:22:10 amypond kernel: [    9.981982]  ? copy_mount_options+0xb8/0x140
Jul  7 00:22:10 amypond kernel: [    9.981985]  ? SyS_mount+0x7e/0xd0
Jul  7 00:22:10 amypond kernel: [    9.981992]  ? entry_SYSCALL_64_fastpath+0x17/0x98
Jul  7 00:22:10 amypond kernel: [    9.981994] Code: 83 c4 50 5b 41 5a 41 5c 41 5d 41 5e 5d c3 48 89 df 89 45 8c e8 24 a5 fa ff 8b 45 8c 48 83 c4 50 5b 41 5a 41 5c 41 5d 41 5e 5d c3 <0f> ff eb c8 b8 f4 ff ff ff eb c9 0f 1f 80 00 00 00 00 0f 1f 44
Jul  7 00:22:10 amypond kernel: [    9.982070] ---[ end trace ea3dfd14241094c0 ]---
```

**Comment by behlendorf** [comment]
Thanks!  @tonyhutter let's get this fix folded in to 0.6.5.11.

**Comment by tonyhutter** [comment]
I added this to the 0.6.5.11 patch list.  I can confirm it fixes the warning on Ubuntu 17.04

**Comment by tonyhutter** [comment]
Fix was included in 0.6.5.11, which is now out.  Closing
